### DISCUSSION TITLE
send message not sending

### DISCUSSION THREAD

**Discussion by anggatinggalklik** [tulir/whatsmeow#715] [General]
{"type": "whatsmeow.SendResponse", "value": {"Timestamp":"2024-12-19T12:45:29Z","ID":"3EB0929206976A96095C03","ServerID":0,"DebugTimings":{"Queue":583,"Marshal":160000,"GetParticipants":0,"GetDevices":234368667,"GroupEncrypt":0,"PeerEncrypt":473123500,"Send":164250,"Resp":239223666,"Retry":0}}}

When sending messages via WhatsApp API using WhatsMeow, the message is not sent even though no errors appear. The response received shows SendResponse data which seems successful, but the message does not reach the recipient and sender.

have respon from sendmessage func
{
  "type": "whatsmeow.SendResponse",
  "value": {
    "Timestamp": "2024-12-19T12:45:29Z",
    "ID": "3EB0929206976A96095C03",
    "ServerID": 0,
    "DebugTimings": {
      "Queue": 583,
      "Marshal": 160000,
      "GetParticipants": 0,
      "GetDevices": 234368667,
      "GroupEncrypt": 0,
      "PeerEncrypt": 473123500,
      "Send": 164250,
      "Resp": 239223666,
      "Retry": 0
    }
  }
}

**Comment by ghost** [comment]
What message ?

**Comment by anggatinggalklik** [comment]
![Screenshot_20241220-112344](https://github.com/user-attachments/assets/0a7ae385-c34b-4e36-9ff8-32386f29b573)

in receiver has look this

**Comment by edosssa** [comment]
same issue

**Comment by ghost** [comment]
**Common problems:**  

- **Missed prekey exchange**  
  - Caused by the client stopping immediately  
  - ...

- **Wrong message protocol**  
  - Some required fields were not set correctly
  - ...

**Reply by edosssa** [reply]
Prior I would disconnect the clients and only connect them when I want to use them (send a message). I have since stopped that approach. Could that by why and do I need to logout the old sessions and login again?

**Reply by edosssa** [reply]
I am also seeing ```Failed to check if xxx's identity is trusted: sql: no rows in result set``` a lot in the console

**Reply by ghost** [reply]
> Prior I would disconnect the clients and only connect them when I want to use them (send a message). I have since stopped that approach. Could that by why and do I need to logout the old sessions and login again?

> I am also seeing `Failed to check if xxx's identity is trusted: sql: no rows in result set` a lot in the console

Re-logging in may resolve these problems. Every time a new device login, it gets its own ID (<num>:<device_id>@s.whatsapp.net) and some unique identities, which trigger a new prekey exchange with another number.

**Reply by edosssa** [reply]
Logging out seemed to do the trick for me. For what it's worth I'd recommend that anyone using this has to keep the socket connection alive all the time to avoid a situation like this
